### DISCUSSION TITLE
Launch a job with srun?

### DISCUSSION THREAD

**Discussion by elyurn** [atmos-cloud-sim-uj/numba-mpi#154] [Q&A]
I have a small script

```python
import numba_mpi
print("Hello! I m rank %d from %d running in total..." % (numba_mpi.rank(), numba_mpi.size()))
```
that work well with `mpiexec python -m myscript`: it shows me two different ranks out of two running in total.

However, using srun, it doesn't work: `srun -q parallel -N 2 python -m myscript` **while it works in `mpi4py`**. I prints the rank 0 twice.

How can I make it work?

**Comment by elyurn** [comment]
I had a problem with multiple installed versions of mpi4py, my bad. Everything works fine.

**Reply by slayoo** [reply]
Thanks for reporting back. One of the design goals of `numba-mpi` is to be interoperable with `mpi4py` - so that you can use both in one program, that is why `numba-mpi` `dlopen`s the same shared lib as `mpi4py`, but indeed in case of multiple `mpi4py`s things might be broken...
